### ISSUE TITLE
speed is slow in webcam

### ISSUE THREAD

**Issue by Digbijay2** [ildoonet/tf-pose-estimation#352]
hello,
wen i am running the code run_webcam.py then my webcam opens but it takes a long time time to process every frame due to which seems to run slowly.
please help someone to recover this.
i am using Intel i3 processor with gpu of Intel® HD Graphics 5500

**Comment by Lynya** [comment]
I think you need a nvidia gpu like at least gtx 1060.

**Comment by harrisonford** [comment]
Yes, makes sense that an integrated hd graphics runs slow. Sorry @Digbijay2 but machine learning stuff is really GPU or TPU intensive!

If you lack the hardware maybe I advice checking google cloud or another similar service.

**Comment by MuhammadAsadJaved** [comment]
Hi, I am using GPU but speed is very slow,  does i need to make some changes in run_webcam.py file so that it can use GPU or by default it is using GPU?

if i need to define that it should use GPU then what changes i should make? please share  code. 

this is my code. 

...................................................................................................................................................................
import logging
import time

import cv2

from tf_pose.estimator import TfPoseEstimator

logger = logging.getLogger('TfPoseEstimator-WebCam')
logger.setLevel(logging.DEBUG)
ch = logging.StreamHandler()
ch.setLevel(logging.DEBUG)
formatter = logging.Formatter('[%(asctime)s] [%(name)s] [%(levelname)s] %(message)s')
ch.setFormatter(formatter)
logger.addHandler(ch)

fps_time = 60
resize_out_ratio=4.0
model='mobilenet_thin'
get_graph_path='/home/asad/t_f_p/tf_pose/graph_opt.pb'
if __name__ == '__main__':

    logger.debug('initialization %s : %s' % (model, get_graph_path))
    w, h = (432, 368)
    e = TfPoseEstimator(get_graph_path, target_size=(432, 368))
    logger.debug('cam read+')
    cam = cv2.VideoCapture(0)
    ret_val, image = cam.read()
    logger.info('cam image=%dx%d' % (image.shape[1], image.shape[0]))

    while True:
        ret_val, image = cam.read()

        logger.debug('image process+')
        humans = e.inference(image, resize_to_default=(w > 0 and h > 0), upsample_size=resize_out_ratio)

        logger.debug('postprocess+')
        image = TfPoseEstimator.draw_humans(image, humans, imgcopy=False)

        logger.debug('show+')
        cv2.putText(image,
                    "FPS: %f" % (1.0 / (time.time() - fps_time)),
                    (10, 10),  cv2.FONT_HERSHEY_SIMPLEX, 0.5,
                    (0, 255, 0), 2)
        cv2.imshow('tf-pose-estimation result', image)
        fps_time = time.time()
        if cv2.waitKey(1) & 0xFF == ord('q'):
            break
        logger.debug('finished+')

    cv2.destroyAllWindows()

**Comment by gkhan496** [comment]
No, you don't need to do anything. I used cuda9, cudnn7.4.1 and tensorflow-gpu 1.9 (least).

**Comment by MuhammadAsadJaved** [comment]
OK thank you very much bro.  @gkhan496

**Comment by mhmmdgamal** [comment]
i use gpu and install tensorflow gpu 1.13 and still slow 
it log that it use cpu
